lib/last_login_sync/gateway/set_last_login.rb

```diff
@@ -0,0 +1,10 @@
+class LastLoginSync::Gateway::SetLastLogin
+  def set(date:, usernames:)
+    usernames.each_slice(100) do |_username_slice|
+      User
+        .where([[:username, usernames]])
+        .where { (last_login =~ nil) | (last_login < date) }
+        .update(last_login: date)
+    end
```

Comment:
I was imagining this part of the update not knowing about the date logic.
Any user that we found gets last_login updated to now. 

User
        .where([[:username, usernames]])
        .update(last_login: Date.today)

Reply:
This code shouldn't be responsible for choosing the date.
The date to check + update to should be at a higher level. In this case, the task.